src/vs/workbench/services/positronConsole/browser/positronConsoleService.ts

```diff
@@ -2332,15 +2372,23 @@ class PositronConsoleInstance extends Disposable implements IPositronConsoleInst
 	 * Sets pending code.
 	 * @param pendingCode The pending code to set.
 	 */
-	setPendingCode(pendingCode?: string) {
+	setPendingCode(pendingCode?: string, executionId?: string) {
+		// If we have both pending code and an execution ID, store it for later use
+		if (pendingCode && executionId) {
+			this._pendingExecutionIds.set(pendingCode, executionId);
```

Comment:
I'm not too familiar with this code, but don't we maintain a single value of `pendingCode` and if so, could we maintain a single `_pendingExecutionId`?

Reply:
You've found an edge case that is not currently handled. :-( Right now all the pending code gets rolled into a single item that gets executed at once. This basically means that if you attempt to use the API to queue up multiple executions they won't resolve correctly. 

See #1220.